packages/dx-grid-core/src/plugins/integrated-summary/computeds.js

```diff
@@ -27,18 +27,12 @@ const rowsSummary = (rows, summaryItems, getCellValue, calculator) => summaryIte
     return acc;
   }, []);
 
-export const totalSummaryValues = (
-  rows,
-  summaryItems,
-  getCellValue,
-  getRowLevelKey,
-  isGroupRow,
-  getCollapsedRows,
-  calculator = defaultSummaryCalculator,
-) => {
-  const plainRows = rows.reduce((acc, row) => {
+const expandRows = (
+  rows, getRowLevelKey, getCollapsedRows, isGroupRow, includeGroupRow = false,
+) => rows
+  .reduce((acc, row) => {
     if (getRowLevelKey && getRowLevelKey(row)) {
-      if (!isGroupRow || !isGroupRow(row)) {
+      if (includeGroupRow || !(isGroupRow && isGroupRow(row))) {
```

Comment:
The `includeGroupRow` and `isGroupRow && isGroupRow(x)` should be [moved out of cycle](https://en.wikipedia.org/wiki/Loop-invariant_code_motion).

```js
const shouldIncludeGroupRow = includeGroup || !isGroupRow ? () => true : row => !isGroupRow(row);
...
if (shouldIncludeGroupRow(row)) {
  acc.push(row);
}
```

Same applies for other neighbor checks.